tests/unit_tests/test_client.py

```diff
@@ -225,12 +304,15 @@ def test_uc_lifecycle(unleash_client):
         status=200,
         headers={"etag": "W/somethingelse"},
     )
-    time.sleep(REFRESH_INTERVAL * 2)
+    fetch_signal.clear()
+    fetch_signal.wait(timeout=REFRESH_INTERVAL * 3)
```

Comment:
Gotta wait for the first fetch (304), then the second fetch (200), then we give the lock another increment as a grace period - so REFRESH * 3

Reply:
I want to say it's clearer when you have the whole test in front of you. I'd like to actually just throw this test away at some point. It's really not a very clear test